homeassistant/components/nanoleaf/config_flow.py

```diff
@@ -0,0 +1,226 @@
+
+USER_SCHEMA: Final = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Nanoleaf config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize a Nanoleaf flow."""
+        self.nanoleaf: Nanoleaf
+
+        # For discovery integration import
+        self.discovery_conf: dict
+        self.device_id: str
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle Nanoleaf flow initiated by the user."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=USER_SCHEMA, last_step=False
+            )
+        self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+        self.nanoleaf = Nanoleaf(user_input[CONF_HOST])
+        try:
+            await self.hass.async_add_executor_job(self.nanoleaf.authorize)
+        except Unavailable:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=USER_SCHEMA,
+                errors={"base": "cannot_connect"},
+                last_step=False,
+            )
+        except NotAuthorizingNewTokens:
+            pass
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unknown error connecting to Nanoleaf")
+            return self.async_show_form(
+                step_id="user",
+                data_schema=USER_SCHEMA,
+                last_step=False,
+                errors={"base": "unknown"},
+            )
+        return self.async_show_form(step_id="link")
+
+    async def async_step_zeroconf(
+        self, discovery_info: DiscoveryInfoType
+    ) -> FlowResult:
+        """Handle Nanoleaf Zeroconf discovery."""
+        _LOGGER.debug("Zeroconf discovered: %s", discovery_info)
+        return await self._async_discovery_handler(discovery_info)
+
+    async def async_step_homekit(self, discovery_info: DiscoveryInfoType) -> FlowResult:
+        """Handle Nanoleaf Homekit discovery."""
+        _LOGGER.debug("Homekit discovered: %s", discovery_info)
+        return await self._async_discovery_handler(discovery_info)
+
+    async def _async_discovery_handler(
+        self, discovery_info: DiscoveryInfoType
+    ) -> FlowResult:
+        """Handle Nanoleaf discovery."""
+        host = discovery_info["host"]
+        name = discovery_info["name"].replace(f".{discovery_info['type']}", "")
+        await self.async_set_unique_id(name)
+        self._abort_if_unique_id_configured({CONF_HOST: host})
+        self.nanoleaf = Nanoleaf(host)
+
+        # Import from discovery integration
+        self.device_id = discovery_info["properties"]["id"]
+        self.discovery_conf = cast(
+            dict,
+            await self.hass.async_add_executor_job(
+                load_json, self.hass.config.path(CONFIG_FILE)
```

Comment:
We should migrate this to use `homeassistant.helpers.storage` in a future PR

Reply:
The token is now stored in the config entry. The file is deleted when all discovery devices have been imported.

Reply:
I missed your `os.remove`.  I'm used to seeing `os.unlink` 👍